### PR TITLE
Fix source path in Lambda distrib

### PR DESCRIPTION
**Pull request by rdettai**
### Description

Bug introduced by https://github.com/quickwit-oss/quickwit/pull/5148

Also removed the overwrite option which was hardcoded and buggy anyway.

### How was this PR tested?

Added an end 2 end integration test exposed in the Makefile. Note that the test is currently failing until https://github.com/quickwit-oss/quickwit/pull/5330 is also merged.

### CODE REVIEW

**Comment by rdettai** [comment]
Internal CI run: https://github.com/quickwit-inc/terraform/actions/runs/10455107621

File: quickwit/quickwit-lambda/src/indexer/model.rs [hunk 1]
```diff
@@ -38,6 +38,7 @@ impl IndexerEvent {
             IndexerEvent::S3(event) => [
                 "s3://",
                 event.records[0].s3.bucket.name.as_ref().unwrap(),
+                "/",
```
Comment (trinity-1686a) [inline quickwit/quickwit-lambda/src/indexer/model.rs:41]: it feels like the kind of things `test_s3_event_uri` should have caught. What gives?

**Review by trinity-1686a — APPROVED** [review]

File: quickwit/quickwit-lambda/src/indexer/model.rs (hunk 1 above)
Comment (rdettai) [reply quickwit/quickwit-lambda/src/indexer/model.rs:41]: I added a make command to run the Lambda tests, but these tests are not running in CI (for performance reasons)

Comment (trinity-1686a) [reply quickwit/quickwit-lambda/src/indexer/model.rs:41]: okay. then we should make sure to run them manually when they may be relevant

**Pull request merged**